tools/azsdk-cli/README.md

```diff
@@ -1,56 +1,138 @@
+To build the project:
+
+```sh
+dotnet build
+```
+
+### Run
+
+To run the CLI locally:
+
+```sh
+dotnet run --project Azure.Sdk.Tools.Cli -- --help
+```
+
+### Test
+
+To run the tests:
+
+```sh
+dotnet test
 ```
 
-My vision here is that these individual tools should be:
-- implemented as standalone servers
-- when DI-ed into our `hub` server, provide a `route` that can be provided by the hub server
+## Project Structure and Information
 
-This way, users can add local `mcp`
+This project is both a [System.CommandLine](https://learn.microsoft.com/en-us/dotnet/standard/commandline/) app and an [MCP server](https://github.com/modelcontextprotocol/csharp-sdk).
 
-|Pro/Con|Description|
-|---|---|
-|✅|Tools will work standalone for easy development and testing.|
-|✅|Much more granular control over what versions of which tools are present in the hub.|
-|🟥|Users will have to match a server convention at the very least, and implement an interface that makes their actual workload pluggable into the `hub`.|
-|🟥|EngSys will need to add good integration testing of the contract, so that local development -> deployment in `everything server` will never surprise them.|
+### Directory Structure
 
-## Additional features
+- *Azure.Sdk.Tools.Cli* - Core project for cli/mcp logic
+    - *Commands* - Shared classes for CLI commands
+    - *Configuration* - Constants and other classes
+    - *Helpers* - Helper logic for parsing data
+    - *Models* - Shared models, response classes and schemas
+    - *Services* - Helper classes for working with upstream services, e.g. azure, devops, github
+    - *Tools* - CLI commands and MCP tool implementations
+- *Azure.Sdk.Tools.Cli.Tests* - Test project
+- *Azure.Sdk.Tools.Cli.Contract* - Common classes/interfaces
+- *Azure.Sdk.Tools.Cli.Analyzer* - Compilation addons to enforce conventions
+    - Enforce all tools handled by try/catch
+    - Enforce tool inclusion in service registration for dependency injector
 
-Concept of `chaining` of various MCP servers. Can this hub support taking the output of one spoke and shoving it to another? We probably should right?
+### Pipelines
 
+Public CI - https://dev.azure.com/azure-sdk/public/_build?definitionId=7677
 
-## Some random DOs and DON'Ts of this server
+Release - https://dev.azure.com/azure-sdk/internal/_build?definitionId=7684 
+
+## Adding a New Tool
+
+Tool classes are the core of the azure sdk app and implement CLI commands or MCP server tools. To add a new tool start with the following:
+
+- Determine the right directory and tool class name under `Azure.Sdk.Tools.Cli/Tools`
+- Reference or copy from the [hello world tool](Azure.Sdk.Tools.Cli/Tools/HelloWorldTool/HelloWorldTool.cs) for an example.
+- See [Tools/README.md](./Azure.Sdk.Tools.Cli/Tools/README.md) for more details
+
+Each tool must implement a `GetCommand()` method and a `HandleCommand(...)` method. These allow the tool to be called from CLI mode.
```

Comment:
I assume GetCommand is for the CLI portion, correct? Is it a must to implement if you only need the MCP Functionality?

Reply:
Are there any MCP tool areas we have right now that we wouldn't also want to expose in the CLI?

Right now we expose some tools to the MCP server and not the CLI, but they still live within a classes that have functionality exposed to both. We could split this up but we would need to define another base class that doesn't require it.

FYI @scbedd

Reply:
Yeah, we would simply create a more base base-class that didn't implement `GetCommand()`. We will still discover _those_ via our static analyzer to ensure it's added to the tools list, but it wouldn't force a CLI portion.

I do think it's quite valuable to force the CLI side (pit of success) but I won't die on that hill 👍

Reply:
Ok let's have the default be add both CLI and MCP tool together, and if we find any cases where that isn't feasible or realistic then we can consider our options.

Reply:
Can we reword this to match what Wes said? by default we expect this behavior, but if it doesn't apply, make sure to communicate the reasons